app/src/main/java/com/example/triptracker/view/profile/UserProfileEditScreen.kt

```diff
@@ -0,0 +1,526 @@
+import androidx.compose.material.Text
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Add
+import androidx.compose.material.icons.filled.CalendarMonth
+import androidx.compose.material.icons.outlined.Image
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.DatePicker
+import androidx.compose.material3.DatePickerDefaults
+import androidx.compose.material3.DatePickerDialog
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.FilledTonalButton
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.OutlinedTextFieldDefaults
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.rememberDatePickerState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.navigation.compose.rememberNavController
+import coil.compose.AsyncImage
+import com.example.triptracker.model.profile.UserProfile
+import com.example.triptracker.view.Navigation
+import com.example.triptracker.view.NavigationBar
+import com.example.triptracker.view.theme.Montserrat
+import com.example.triptracker.view.theme.md_theme_grey
+import com.example.triptracker.view.theme.md_theme_light_dark
+import com.example.triptracker.view.theme.md_theme_light_error
+import com.example.triptracker.view.theme.md_theme_orange
+import java.time.Instant
+import java.time.LocalDate
+import java.time.ZoneId
+import java.time.format.DateTimeFormatter
+import java.util.Date
+
+/**
+ * This function retrieves the user profile from the database.
+ *
+ * @return user profile.
+ */
+private fun retrieveProfile(): UserProfile {
+  // TODO: implement the retrieval of the ambient user profile
+  return UserProfile(
+      "jean.rousseau@epfl.ch",
+      "Jean-Jacques",
+      "Rousseau",
+      Date(),
+      "DarkSkyMan",
+      "profileImageUrl",
+      listOf(),
+      listOf())
+}
+
+/**
+ * This function saves the user profile to the database.
+ *
+ * @param profile : user profile to be saved.
+ */
+private fun updateProfile(
+    profile: UserProfile,
+    name: String,
+    surname: String,
+    mail: String,
+    birthdate: LocalDate,
+    username: String,
+    imageUrl: String?
+) {
+  // TODO: Implement saving the profile to the database
```

Comment:
You can easily do it using the updateProfileInDb() function from the UserProfileViewModel class
(This function is already functioning) :)

Reply:
Alright thank you ! For now I add the command but I leave it commented, because as I am not retrieving the ambiant user profile this could lead to an error (by wanting to modify a profile that doesn't exist).